### ISSUE TITLE
RuntimeError: dictionary changed size during iteration

### ISSUE THREAD

**Issue by repoocaj** [ibm-watson-iot/iot-python#95]
I just updated to 0.3.3 and I'm getting this trackback in my application:

```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5.2/lib/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.5.2/lib/python3.5/threading.py", line 862, in run
    self._target(*self._args, **self._kwargs)
  File "/home/cwhitcher/venvs/python352/lib/python3.5/site-packages/paho/mqtt/client.py", line 2650, in _thread_main
    self.loop_forever(retry_first_connection=True)
  File "/home/cwhitcher/venvs/python352/lib/python3.5/site-packages/paho/mqtt/client.py", line 1481, in loop_forever
    rc = self.loop(timeout, max_packets)
  File "/home/cwhitcher/venvs/python352/lib/python3.5/site-packages/paho/mqtt/client.py", line 1003, in loop
    rc = self.loop_read(max_packets)
  File "/home/cwhitcher/venvs/python352/lib/python3.5/site-packages/paho/mqtt/client.py", line 1284, in loop_read
    rc = self._packet_read()
  File "/home/cwhitcher/venvs/python352/lib/python3.5/site-packages/paho/mqtt/client.py", line 1849, in _packet_read
    rc = self._packet_handle()
  File "/home/cwhitcher/venvs/python352/lib/python3.5/site-packages/paho/mqtt/client.py", line 2311, in _packet_handle
    return self._handle_connack()
  File "/home/cwhitcher/venvs/python352/lib/python3.5/site-packages/paho/mqtt/client.py", line 2372, in _handle_connack
    self.on_connect(self, self._userdata, flags_dict, result)
  File "/home/cwhitcher/venvs/python352/lib/python3.5/site-packages/ibmiotf/application.py", line 272, in on_connect
    for subscription in self._subscriptions:
RuntimeError: dictionary changed size during iteration
```
I think there needs to be a lock around access to self._subscriptions since it can be modified from multiple threads.

I'll try it out and report back.

**Comment by durera** [comment]
Addressed in e36b873a201473c422234ec31af38b1415353af8

**Comment by durera** [comment]
This should be fixed in 0.3.4 (released Feb 25th) so closing this issue now -- https://github.com/ibm-watson-iot/iot-python/releases/tag/0.3.4
